### ISSUE TITLE
Cancel workflow leads to `Unexpected end of JSON input`

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal-web#213]
I'm trying to Cancel a workflow and get shown:

There has been an error, reason:
"Unexpected end of JSON input"

<img width="1326" alt="Screenshot 2023-06-23 at 14 33 12" src="https://github.com/fractal-analytics-platform/fractal-web/assets/18033446/d3ded317-bc9d-4f19-8489-411807fec3a4">

From fractal-web log:
```
Action: stop
Requesting stop job execution from server
SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at parseJSONFromBytes (/opt/fmi/gliberal/fractal/20230619_deployment/fractal-web/node_modules/undici/lib/fetch/body.js:572:15)
    at successSteps (/opt/fmi/gliberal/fractal/20230619_deployment/fractal-web/node_modules/undici/lib/fetch/body.js:512:23)
    at /opt/fmi/gliberal/fractal/20230619_deployment/fractal-web/node_modules/undici/lib/fetch/util.js:814:56
    at node:internal/process/task_queues:141:7
    at AsyncResource.runInAsyncScope (node:async_hooks:202:9)
    at AsyncResource.runMicrotask (node:internal/process/task_queues:138:8)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

Console doesn't show much more

**Comment by tcompa** [comment]
Can you check also fractal-server logs?

**Comment by jluethi** [comment]
Yes, saw nothing showing up in the .err log, surprisingly. Will check again

**Comment by jluethi** [comment]
At least on the INFO level, nothing shows up in the .err or .out file about it

**Comment by tcompa** [comment]
I debugged this a bit further, and the actual reason is that the stop-job endpoint does not return any response body, see [here](https://github.com/fractal-analytics-platform/fractal-server/blob/681d9173dec0707fd40a5fb98b71cc6b6fbfe8b4/fractal_server/app/api/v1/job.py#L134-L174):
```python
@router.get(
    "/project/{project_id}/job/{job_id}/stop/",
    status_code=200,
)
async def stop_job(
    project_id: int,
    job_id: int,
    user: User = Depends(current_active_user),
    db: AsyncSession = Depends(get_db),
) -> Optional[ApplyWorkflow]:
    """
    Stop execution of a workflow job (only available for slurm backend)
    """

    # This endpoint is only implemented for SLURM backend
    settings = Inject(get_settings)
    backend = settings.FRACTAL_RUNNER_BACKEND
    if backend != "slurm":
        raise HTTPException(
            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
            detail=(
                "Stopping a job execution is not implemented for "
                f"FRACTAL_RUNNER_BACKEND={backend}."
            ),
        )

    # Get job from DB
    output = await _get_job_check_owner(
        project_id=project_id,
        job_id=job_id,
        user_id=user.id,
        db=db,
    )
    job = output["job"]

    # Write shutdown file
    shutdown_file = Path(job.working_dir) / SHUTDOWN_FILENAME
    with shutdown_file.open("w") as f:
        f.write(f"Trigger executor shutdown for {job.id=}, {project_id=}.")

    return Response(status_code=status.HTTP_200_OK)
```
for this reason, the following block in `src/lib/server/api/v1/monitoring_api.js`
```
	if (response.ok) {
		return await response.json();
	}
```
fails, because it cannot find the correct JSON content.

We could find a quick fix on fractal-web side (e.g. not using `.json()`, but simply creating an empty response), but perhaps this could create issues in cases where the API call fails.

I think the best is to comply with the expected return type in the stop-job fractal-server endpoint (that is, we should return the job object). This is an easy fix, which we can release as part of fractal-server 1.3.1 very rapidly.

**Comment by jluethi** [comment]
Makes sense, sounds like a good plan to solve this with a fractal-server update :)

**Comment by tcompa** [comment]
This is in principle fixed by switching to fractal-server 1.3.1 (real-life tests are welcome).

**Comment by tcompa** [comment]
This is working correctly for fractal-web, see fractal-server for an additional issue:
* https://github.com/fractal-analytics-platform/fractal-server/issues/766

Closing.
